rest_framework/renderers.py

```diff
@@ -72,7 +72,7 @@ def get_indent(self, accepted_media_type, renderer_context):
             # If the media type looks like 'application/json; indent=4',
             # then pretty print the result.
             # Note that we coerce `indent=0` into `indent=None`.
-            base_media_type, params = parse_header(accepted_media_type.encode('ascii'))
+            base_media_type, params = parse_header_params(accepted_media_type, encoding='ascii')
```

Comment:
Here the value returned is immediately converted to `int`, nobody cares if it's `bytes` or `str` 
```suggestion
            base_media_type, params = parse_header_params(accepted_media_type.encode('ascii'))
```

Reply:
so, the issue is you can't directly pass in bytes to the method [parse_header_parameters](https://github.com/django/django/blob/eb3699ea775548a22e0407ad12bf8cbdeaf95ff5/django/utils/http.py#L400-L427) since it works with strings. the earlier method [parse_header](https://github.com/django/django/blob/9cf2564d38751a46a71fd322cc08bda717f452a4/django/http/multipartparser.py#L715-L744) worked with bytes.

hence passing the encoding format as well to `parse_header_params` so that it decodes in the same format when returning the value.

same reason, for the change everywhere else.

Reply:
Right, my bad
```suggestion
            base_media_type, params = parse_header_params(accepted_media_type)
```
If it's all strings anyway, we don't need to worry about encoding here at all.
Similarly elsewhere -- Django used to want these as `bytes`, so DRF, which had them as `str`, encoded them for Django's benefit. But if Django takes them as `str`, and (as I've shown), either the return type is unimportant or the return value is turned back to `str` immediately -- then DRF can simply drop the code that deals with encoding.

Reply:
for the older versions, the encoding and decoding part needs to remain. as of now, i don't think what you're suggesting would be backwards compatible, unless i'm not getting something wrong.

Reply:
What I'm suggesting is that the compat part look like so:
```python
if django.VERSION > (4, 1):
    from django.utils.http import parse_header_parameters

else:
    from django.http.multipartparser import parse_header

    # Old Django function preferred to take bytes in and return
    # bytes in the pdict values. We humor it.
    def parse_header_parameters(params):
        key, pdict = parse_header(params.encode())
        pdict = {k:v.decode() for k, v in pdict.items()}
        return key, pdict
```
and callers, which all hold the `params` line in a `str` to begin with, and (as I've detailed) at worst need minor changes to handle the `pdict` values returned as `str`, be fixed to forget about encodings.

At first this may seem not-backwards-compatible, because it "misses" the part where different encodings are expected in different parts of the code. But remember, in all the DRF code, the `params` line here has already been decoded and is given to us as a `str`. All the specific encodings achieve is a very odd and obscure "validity test" -- if, say, the `accepted_media_type` in `get_indent()` includes a non-ascii character, and somehow this has not blocked processing earlier, then, with the new version, processing will succeed, where in the old version, a `UnicodeEncodeError` would be raised.

I submit that this is not a very important behavior to preserve, and it is much more important to simplify the code and keep the ideal where, when support of the old version is no longer required, you just remove the shim and change the imports.